action-publish-image/action.yml

```diff
@@ -0,0 +1,69 @@
+name: Build and Push Image
```

Comment:
nit about folder organization: Instead of having a top-level `action-publish-image`, could we have a `github-actions/publish-image/` directory structure?

That way we have a place to put other future actions we write here

Reply:
since we already have `action-publish-code` on top-level, maybe we can move both in a follow up?

Reply:
yep, sounds good, i'll move them in follow-up!